_posts/2022-10-10-snapshot-management.md

```diff
@@ -0,0 +1,158 @@
+
+## SM in action
+
+Think of SM as having your index periodically take selfies. 
+
+### How to set up SM
+
+You can set up SM either using the [REST API](https://opensearch.org/docs/latest/opensearch/snapshots/sm-api/) or through [OpenSearch Dashboards](https://opensearch.org/docs/latest/opensearch/snapshots/sm-dashboards/). Here is the sequence that you should follow.
+
+First, create a repository to hold your snapshots.
+
+Then, [create an SM policy](https://opensearch.org/docs/latest/opensearch/snapshots/sm-api#create-or-update-a-policy). The SM policy describes the following:
+- The indexes for which to take snapshots.
+- How often and when to take snapshots (for example, every hour or every Sunday at midnight).
+- Which snapshots to retain (for example, retain snapshots taken within the last 30 days or retain the last 100 snapshots). This is optional. If you don't specify a retention schedule, OpenSearch will keep all snapshots.
+- How you want to be notified of snapshot events and what snapshot events interest you (creation, deletion, or failure). This is optional. 
+
+You need to set the schedule for your automatic snapshots using a [cron expression](https://opensearch.org/docs/latest/monitoring-plugins/alerting/cron/). Let's dissect the expression `0 8 * * *`, which we'll use in the example below.
+
+:--- | :--- | :--- | :--- | :---
+0 | 8 | * | * | *
+minute| hour | day of month | month | day of year
+
+This cron expression means you want to take snapshots at 8:00 AM every day.
+
+Here's a sample SM policy that puts together all of this information:
+
+```json
+POST _plugins/_sm/policies/daily-policy
+{
+  "description": "Daily snapshot policy",
+  "creation": {
+    "schedule": {
+      "cron": {
+        "expression": "0 8 * * *",
+        "timezone": "UTC"
+      }
+    },
+    "time_limit": "1h"
+  },
+  "deletion": {
+    "schedule": {
+      "cron": {
+        "expression": "0 1 * * *",
+        "timezone": "America/Los_Angeles"
+      }
+    },
+    "condition": {
+      "max_age": "7d",
+      "max_count": 21,
+      "min_count": 7
+    },
+    "time_limit": "1h"
+  },
+  "snapshot_config": {
+    "date_format": "yyyy-MM-dd-HH:mm",
+    "timezone": "America/Los_Angeles",
+    "indices": "*",
+    "repository": "s3-repo",
+    "ignore_unavailable": "true",
+    "include_global_state": "false",
+    "partial": "true",
+    "metadata": {
+      "any_key": "any_value"
+    }
+  },
+  "notification": {
+    "channel": {
+      "id": "NC3OpoEBzEoHMX183R3f"
+    },
+    "conditions": {
+      "creation": true,
+      "deletion": false,
+      "failure": false,
+      "time_limit_exceeded": false
+    }
+  }
+}
+```
+
+You can also set up your SM policy through OpenSearch Dashboards as follows:
+
+<img src="/assets/media/blog-images/2022-10-10-snapshot-management/Snapshot-management-demo.gif" alt="SM Policy Create Workflow" width="150" style="float: left; margin-right: 15px;"/>
```

Comment:
@kolchfa-aws @krisfreedain how do we check this gif is rendering properly on the website?

Reply:
@kolchfa-aws @setiah gif looks way too small:
![Screen Shot 2022-10-07 at 10 47 55 AM](https://user-images.githubusercontent.com/4873623/194619707-e0f6604b-e5a1-471e-b21e-a35de95b619d.png)


I recommend changing line 129 to:

`<img src="/assets/media/blog-images/2022-10-10-snapshot-management/Snapshot-management-demo.gif" alt="SM Policy Create Workflow"/>{: .img-fluid }`

Reply:
Thank you, Kris! I will change that.

Reply:
Excellent - thanks @kolchfa-aws - I'll push this live on Monday morning.